### DISCUSSION TITLE
注册resource后渲染组件，菜单执行关闭已保存，提示resource 无provider

### DISCUSSION THREAD

**Discussion by yinyinfurong8888** [opensumi/core#1738] [Q&A] [answered]
注册资源打开方式
```
resourceService.registerResourceProvider({
      scheme: ET_CAUSE_REPORT_SCHEME,
      provideResource: async (uri: URI): Promise<IResource<any>> => {
        const iconClass = this.iconService.fromString('$(report)');
        return {
          uri,
          name: '报告',
          icon: iconClass!,
        };
      },
    });
```
注册渲染组件并设置resource 的默认打开方式为此渲染组件
```
    registry.registerEditorComponent({
      uid: save_id,
      scheme: ET_CAUSE_SAVE_AS_SCHEME,
      component: EtTcmEditorComponent,
      renderMode: EditorComponentRenderMode.ONE_PER_WORKBENCH,
    });

    // 设置resource 的默认打开方式
    registry.registerEditorComponentResolver(ET_CAUSE_SAVE_AS_SCHEME, (resource, results, resolver) =>
      resolver([
        {
          type: 'component',
          componentId: save_id,
        },
      ]),
    );
```
resource可以正常展示，
![image](https://user-images.githubusercontent.com/83274944/192705650-a77ef314-920f-4d97-9038-3f39f852587b.png)

但是在编辑器标题（MenuId.EditorTitle 注册区域），右键菜单渲染关闭已保存，就会触发报错@opensumi/ide-core-common/lib/log.js 报错
![image](https://user-images.githubusercontent.com/83274944/192705855-1466343d-6b58-4aa7-bf7d-0d6213c384a7.png)

![image](https://user-images.githubusercontent.com/83274944/192705984-b0b47250-8094-4cfe-b510-f4da23131c97.png)

pwm_report:/e%3A/code/test/pict/09211.pict 是新的resource.uri

看了下个人代码，没有对保存做处理

**Comment by erha19** [comment]
@yinyinfurong8888 目前这块对功能有影响吗? 预期的表现是啥？

**Comment by erha19** [comment, marked as answer]
@yinyinfurong8888 这里的报错不影响实际功能，将在在 2.20.3 版本中移除该报错 #1739.
